tools/diffscripts/diffapi.ps1

```diff
@@ -0,0 +1,370 @@
+    Write-Host -ForegroundColor Red "Please get the AsmDiff tool from the dotnet arcade by installing it using the following command line:"
+    Write-Host
+    Write-Host -ForegroundColor Yellow "dotnet tool install Microsoft.Dotnet.AsmDiff -g --add-source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json --version 6.0.0-beta.21161.15"
+    Write-Host
+    Write-Host -ForegroundColor Cyan -NoNewline "NOTE: The command above will install the AsmDiff tool " 
+    Write-Host -ForegroundColor Black -BackgroundColor Cyan -NoNewline "globally"
+    Write-Host -ForegroundColor Cyan " and will allow you to run the script without parameters." 
+    Write-Host -ForegroundColor Cyan -NoNewline "NOTE: If you don't want it to be installed globally remove the "
+    Write-Host -ForegroundColor White -NoNewline "-g " 
+    Write-Host -ForegroundColor Cyan  -NoNewline "flag from the above command and specify the tool location with " 
+    Write-Host -ForegroundColor White -NoNewline "-AsmToolExecutable" 
+    Write-Host -ForegroundColor Cyan " parameter when running this script."
+    Write-Host
+    Write-Host -ForegroundColor Yellow "dotnet tool install Microsoft.Dotnet.AsmDiff --add-source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json --version 6.0.0-beta.21161.15 --tool-path c:\tools\asmdiff"
+    Write-Host -ForegroundColor Yellow -NoNewline ".\" 
+    Write-Host -ForegroundColor Yellow $MyInvocation.MyCommand "-AsmToolExecutable c:\tools\asmdiff\dotnet-asmdiff.exe"
+    Write-Host
+    Write-Host -ForegroundColor Cyan "NOTE: This requires .NET Core 2.1 SDK or higher, but it is recommended to use .NET Core 3.1"
+    Write-Host -ForegroundColor Cyan "NOTE: Install .NET Core 3.1 from here: https://dotnet.microsoft.com/download/dotnet/thank-you/sdk-3.1.407-windows-x64-installer"
+    Write-Host -ForegroundColor Cyan "NOTE: dotnet tool install help: https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-tool-install"
+    Write-Host
+
+    $hasAFault = $true
+}
+
+# We can have TWO faults so instead of bailing out after each one we can show both
+if ($hasAFault) 
+{
+    Write-Host -ForegroundColor Yellow "Please correct the above and rerun this tool."
+    exit 1
+}
+
+# Hardcoded list of assembly names
+$assemblyRootNames = @(
+    "Microsoft.Azure.Devices.Shared",
+    "Microsoft.Azure.Devices.Client",
+    "Microsoft.Azure.Devices",
+    "Microsoft.Azure.Devices.Provisioning.Client",
+    "Microsoft.Azure.Devices.Provisioning.Service",
+    "Microsoft.Azure.Devices.Provisioning.Transport.Amqp",
+    "Microsoft.Azure.Devices.Provisioning.Transport.Mqtt",
+    "Microsoft.Azure.Devices.Provisioning.Transport.Http",
+    "Microsoft.Azure.Devices.Provisioning.Security.Tpm"
+)
+
+# All of the files from the build
+$assemblyFilePath = @(
+    ((Join-Path -Path $repoRootPath -ChildPath "\shared\src\bin\Release\netstandard2.1\") + ($assemblyRootNames[0] + ".dll")),
```

Comment:
What came of the `-AdditionalChildPath` parameter for the assembly name?

Reply:
Never mind. I saw your previous comment about it, now.

You'd have to nest two Join-Path statements to be consistent, which I'd still recommend rather than hybrid.